frontend/javascripts/oxalis/model/accessors/dataset_layer_rotation_accessor.ts

```diff
@@ -0,0 +1,414 @@
+import * as THREE from "three";
+import type {
+  AffineTransformation,
+  APIDataLayer,
+  APIDataset,
+  APISkeletonLayer,
+  CoordinateTransformation,
+} from "types/api_flow_types";
+import { mod } from "libs/utils";
+import MultiKeyMap from "libs/multi_key_map";
+import _ from "lodash";
+import memoizeOne from "memoize-one";
+import {
+  createAffineTransformFromMatrix,
+  createThinPlateSplineTransform,
+  chainTransforms,
+  invertTransform,
+  transformPointUnscaled,
+  nestedToFlatMatrix,
+  type Transform,
+} from "../helpers/transformation_helpers";
+import { getLayerByName } from "./dataset_accessor";
+import type BoundingBox from "../bucket_data_handling/bounding_box";
+
+const IDENTITY_MATRIX = [
+  [1, 0, 0, 0],
+  [0, 1, 0, 0],
+  [0, 0, 1, 0],
+  [0, 0, 0, 1],
+] as NestedMatrix4;
+
+export const IDENTITY_TRANSFORM: CoordinateTransformation = {
+  type: "affine",
+  matrix: IDENTITY_MATRIX,
+};
+
+// cf. https://en.wikipedia.org/wiki/Rotation_matrix#In_three_dimensions
+const sinusLocationOfRotationInMatrix = {
+  x: [2, 1],
+  y: [0, 2],
+  z: [1, 0],
+};
+
+const cosineLocationOfRotationInMatrix = {
+  x: [1, 1],
+  y: [0, 0],
+  z: [0, 0],
+};
+
+export const AXIS_TO_TRANSFORM_INDEX = {
+  x: 1,
+  y: 2,
+  z: 3,
+};
+
+export function flatToNestedMatrix(matrix: Matrix4x4): NestedMatrix4 {
+  return [
+    matrix.slice(0, 4) as Vector4,
+    matrix.slice(4, 8) as Vector4,
+    matrix.slice(8, 12) as Vector4,
+    matrix.slice(12, 16) as Vector4,
+  ];
+}
+
+export function getRotationFromTransformation(
+  transformation: CoordinateTransformation | undefined,
+  axis: "x" | "y" | "z",
+) {
+  if (transformation && transformation.type !== "affine") {
+    return 0;
+  }
+  const matrix = transformation ? transformation.matrix : IDENTITY_MATRIX;
+  const cosineLocation = cosineLocationOfRotationInMatrix[axis];
+  const sinusLocation = sinusLocationOfRotationInMatrix[axis];
+  const sinOfAngle = matrix[sinusLocation[0]][sinusLocation[1]];
+  const cosOfAngle = matrix[cosineLocation[0]][cosineLocation[1]];
+  const rotation =
+    Math.abs(cosOfAngle) > 1e-6 // Avoid division by zero
+      ? Math.atan2(sinOfAngle, cosOfAngle)
+      : sinOfAngle > 0
+        ? Math.PI / 2
+        : -Math.PI / 2;
+  const rotationInDegrees = rotation * (180 / Math.PI);
+  // Round to multiple of 90 degrees and keep the result positive.
+  const roundedRotation = mod(Math.round((rotationInDegrees + 360) / 90) * 90, 360);
+  return roundedRotation;
+}
+
+export function getTranslationToOrigin(bbox: BoundingBox): AffineTransformation {
+  const center = bbox.getCenter();
+  const translationMatrix = new THREE.Matrix4()
+    .makeTranslation(-center[0], -center[1], -center[2])
+    .transpose(); // Column-major to row-major
+  return { type: "affine", matrix: flatToNestedMatrix(translationMatrix.toArray()) };
+}
+
+export function getTranslationBackToOriginalPosition(bbox: BoundingBox): AffineTransformation {
```

Comment:
I'd use "fromCenterToOrigin" and "fromOriginToCenter" in the two functions above.